ArduCopter/surface_tracking.cpp

```diff
@@ -71,7 +49,7 @@ bool Copter::SurfaceTracking::get_target_alt_cm(float &_target_alt_cm) const
     if (AP_HAL::millis() - last_update_ms > SURFACE_TRACKING_TIMEOUT_MS) {
         return false;
     }
-    _target_alt_cm = target_dist_cm;
+    target_alt_cm = (copter.pos_control->get_pos_target_z_cm() - copter.pos_control->get_pos_offset_z_cm());
```

Comment:
this is a little tricky actually because it assumes that the caller of surface tracking passed the offset into the position controller.  We might consider reverting the offset state to be within surface tracking so we don't make this assumption.